postgres-appliance/scripts/configure_spilo.py

```diff
@@ -569,11 +570,13 @@ def get_placeholders(provider):
         placeholders.setdefault('USE_WALG_RESTORE', 'true')
     if placeholders.get('WALG_AZ_PREFIX'):
         placeholders.setdefault('USE_WALG_BACKUP', 'true')
+    if all(placeholders.get(n) for n in WALG_SSH_NAMES):
+        placeholders.setdefault('USE_WALG_BACKUP', 'true')
     set_walg_placeholders(placeholders)
 
     placeholders['USE_WALE'] = any(placeholders.get(n) for n in AUTO_ENABLE_WALG_RESTORE +
                                    ('WAL_SWIFT_BUCKET', 'WALE_SWIFT_PREFIX', 'WAL_GCS_BUCKET',
-                                    'WAL_GS_BUCKET', 'WALE_GS_PREFIX', 'WALG_GS_PREFIX'))
+                                    'WAL_GS_BUCKET', 'WALE_GS_PREFIX', 'WALG_GS_PREFIX', 'WALG_SSH_PREFIX'))
```

Comment:
Please add `WALG_SSH_PREFIX` to the [AUTO_ENABLE_WALG_RESTORE](https://github.com/zalando/spilo/pull/549/files#diff-e4dcf964214d58d733ae9874d4a2ab4b8e9eb99213307e1666af5dad21a326bbR37).

And here it could be removed.

Reply:
As I haven't tested it with the AUTO features at all yet, I left it unset. I honestly find spilo_configure to be quite obtuse and difficult to parse.

As I don't use S3 I cannot speak for if it works there or not with this change.  Tests of backing up have been done, but only on small data sets as I'm currently evaluating spilo for our environment, and backing up over SSH is one of our production requirements, so I needed to get this fixed and working first.

I'll follow up with changes as requested once I've tested them, probably later Tuseday.

Reply:
> As I haven't tested it with the AUTO features at all yet, I left it unset. I honestly find spilo_configure to be quite obtuse and difficult to parse.

Sorry about that. It was "growing" for 6 years with mostly ad-hoc changes. That pretty much explains the current state.

> As I don't use S3 I cannot speak for if it works there or not with this change.

That's fine. We have all the tests. First of all intergartion tests with docker-compose: https://github.com/zalando/spilo/tree/master/postgres-appliance/tests, which are execute for every commit, but Minio is used. Also we have end-to-end tests on a real K8s cluster running on AWS. The experience was that end-to-end tests were failing due to the wal-g 0.2.17  not able to communicate with S3, while 0.2.15 was working perfectly fine. If this behavior retain I afraid we can't use the latest version :(

> Tests of backing up have been done

It also would be nice to do a restore, otherwise we are dealing with Schrodinger backup. Actualy this is one of the point of having automated tests where we check different scanarios including backup is created and it is possible to restore from it.

> I'll follow up with changes as requested once I've tested them, probably later Tuseday.

Cool, thank you!

Reply:
Followup:

Testing in progress with the changes requested. I got side-tracked by a build failure in CI when trying to make the container build for us, turns out it was an unrelated capability difference. 

Running a build set and tests. Previous commit at least showed up as working for me.

Unrelated question, but the sysstat package is being installed in the spilo images, and automatically triggered via cron when enabling the backup feature. Is that really intended, or some side-effect of the debian packaging?

Reply:
And now I can confirm that with the requested change ( 9c5c8e1cecda671eab3d78a70869768677111a8b )  it works and can automatically restore  with wal-g as well.

Reply:
> Unrelated question, but the sysstat package is being installed in the spilo images, and automatically triggered via cron when enabling the backup feature. Is that really intended, or some side-effect of the debian packaging?

The `sysstat` is installed only just for convenience, for example one could run `iostat -mx 1` from the container.

Backups are triggered from the cron, yes, but it is not related to the sysstat.

Reply:
Ah, right. Mind if I file a change to turn sysstat on/off as a build argument?

Reply:
We only install a package, but don't start the daemon.
Package itself is quite small, less than 1MB. On top of 720MB this is peanuts.